ironfish/src/mining/stratum/stratumPeers.ts

```diff
@@ -0,0 +1,158 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import net from 'net'
+import { Event } from '../../event'
+import { Config } from '../../fileStores/config'
+import { createRootLogger, Logger } from '../../logger'
+import { SetTimeoutToken } from '../../utils'
+import { StratumServerClient } from './stratumServerClient'
+
+const TWO_HOURS_MS = 2 * 60 * 60 * 1000
+const PEERS_TICK_MS = 10000
+
+export class StratumPeers {
+  readonly logger: Logger
+  readonly maxConnectionsByIp: number
+
+  readonly onBanned = new Event<[StratumServerClient]>()
+
+  banCount = 0
+
+  protected readonly connectionsByIp: Map<string, number> = new Map()
+  protected readonly bannedByIp: Map<string, number> = new Map()
+  protected readonly scoreByIp: Map<string, number> = new Map()
+  protected readonly shadowBans: Set<number> = new Set()
+  protected eventLoopTimeout: SetTimeoutToken | null = null
+
+  constructor(options: { maxConnectionsPerIp?: number; config: Config; logger?: Logger }) {
+    this.logger = options.logger ?? createRootLogger()
+
+    this.maxConnectionsByIp =
+      options.maxConnectionsPerIp ?? options.config.get('poolMaxConnectionsPerIp')
+  }
+
+  start(): void {
+    this.eventLoop()
+  }
+
+  stop(): void {
+    if (this.eventLoopTimeout) {
+      clearTimeout(this.eventLoopTimeout)
+    }
+  }
+
+  punish(client: StratumServerClient, reason?: string, amount = 1): void {
+    if (this.isBanned(client.socket)) {
+      return
+    }
+
+    let banScore = this.scoreByIp.get(client.remoteAddress) ?? 0
+    banScore += amount
+
+    if (banScore < 10) {
+      this.scoreByIp.set(client.remoteAddress, banScore)
+      return
+    }
+
+    this.ban(client, reason)
+    this.scoreByIp.delete(client.remoteAddress)
+  }
+
+  shadowBan(client: StratumServerClient): void {
```

Comment:
This PR removes all occurrences of actually shadowbanning anyone. Do we intend on bringing back the functionality of a shadowban?

Reply:
I think we only ever intend to shadow ban based on version now, and otherwise outright ban them. That should be ok. What do you think?

Reply:
Oh I see, some how the shadow ban call got lost in a rebase.